modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectMetadataFactoryImpl.java

```diff
@@ -189,6 +209,22 @@ protected synchronized void refresh(long time)
 
 			_oidcProviderMetadata = OIDCProviderMetadata.parse(jsonObject);
 
+			if (_allowedIdTokenSigningAlgValues != null) {
+				List<JWSAlgorithm> idTokenJWSAlgs =
+					_oidcProviderMetadata.getIDTokenJWSAlgs();
+
+				Stream<JWSAlgorithm> stream = idTokenJWSAlgs.stream();
+
+				_oidcProviderMetadata.setIDTokenJWSAlgs(
+					stream.filter(
+						jwsAlgorithm ->
+							_allowedIdTokenSigningAlgValues.contains(
```

Comment:
@stian-sigvartsen 
The filtering here does not make sense. The refresh logic here is for re-discover provider metadata info, a filtering here means to filter returned supported algorithms against configured supported algorithms.
A simple test will be if you configure both RS256, HS256 in configuration page, then you will fail algorithm check again.

The filtering place I was refering to is in requestUtil's validate method, where you currently choose header's algorithm to validate.
What we need to do is beforing choosing header's algorithm to validate, we verify header's algorithm against the supported algorithms.